compiler/rustc_hir_analysis/src/check/wfcheck.rs

```diff
@@ -180,8 +180,8 @@ fn check_item<'tcx>(tcx: TyCtxt<'tcx>, item: &'tcx hir::Item<'tcx>) {
         // for `T`
         hir::ItemKind::Impl(ref impl_) => {
             let is_auto = tcx
-                .impl_trait_ref(def_id)
-                .map_or(false, |trait_ref| tcx.trait_is_auto(trait_ref.def_id));
+                .bound_impl_trait_ref(def_id.into())
+                .map_or(false, |trait_ref| tcx.trait_is_auto(trait_ref.skip_binder().def_id));
```

Comment:
this `skip_binder` is correct :+1: we're only looking at the `DefId` which does not change when we substitute generic parameters